src/Discord/Internal/Types/Guild.hs

```diff
@@ -298,6 +301,77 @@ instance FromJSON Role where
          <*> o .: "managed"
          <*> o .: "mentionable"
 
+-- | If there is no such role on the guild return nothing
+--   otherwise return the role. Take the head of the list. List should always be one, because the ID is unique
+roleIdToRole :: Guild -> RoleId -> Maybe Role
+roleIdToRole  g r = bool (Just $ head f) Nothing ( null f)
```

Comment:
Could probably just be `lookup (\x -> roleId x == r) (guildRoles g)`

Reply:
gonna add it!

Reply:
lookup expects [(a,b)] gut guildRoles returns [Role]

Reply:
`find` is what I was thinking!

Reply:
implimented it